src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -234,6 +235,26 @@ public bool TryLoadConfig(
 
                 config = RuntimeConfig;
 
+                // When isDevMode is not null it means we are in a hot-reload scenario, and need to save the previous
+                // mode in the new RuntimeConfig since we do not support hot-reload of the mode.
+                if (isDevMode is not null && config.Runtime is not null && config.Runtime.Host is not null)
```

Comment:
should we rename this var to `incomingRuntimeConfig` since its a little confusing if its the updated one or current one